### ISSUE TITLE
Damage when drawing a cylinder from pdflatex in versions after 2.53

### ISSUE THREAD

**Issue by zarniwhoop73** [vectorgraphics/asymptote#145]
I have not been looking at results from my test documents with enough care. The following MWE produced a boring cylinder up to v2.53. In v2.54 it lost a rectangular part in the top, and also lost the colour, in 2.55 the colour was restored but not the top, similar result through to 2.62. In 2.63, 2.64 and current git the top or far side is restored, but a bigger chunk is taken out from the nearest side and another chunk is missing from the top right corner which eventually drew my attention to it. 

`
\documentclass[12pt]{article}
  
\usepackage{asymptote}

\begin{document}

\begin{asydef}
import three;
\end{asydef}

\begin{center}
\begin{asy}cylinder:
        pdflatex cylinder.tex
        asy -V cylinder-*.asy
        pdflatex cylinder.tex
        pdflatex cylinder.tex

settings.render=16;
size(15cm,0);

draw(unitcylinder, red+opacity(.3));

\end{asy}
\end{center}

\end{document}
`
I use a Makefile withtwo runs of pdflatex after asy because it complained it needed to be rerun :

`
cylinder:
        pdflatex cylinder.tex
        asy -V cylinder-*.asy
        pdflatex cylinder.tex
        pdflatex cylinder.tex

`

[cylinder-2.53.pdf](https://github.com/vectorgraphics/asymptote/files/4357785/cylinder-2.53.pdf)
[cylinder-2.55.pdf](https://github.com/vectorgraphics/asymptote/files/4357786/cylinder-2.55.pdf)
[cylinder-2.63.pdf](https://github.com/vectorgraphics/asymptote/files/4357787/cylinder-2.63.pdf)

**Comment by zarniwhoop73** [comment]
I now see that although current git reported the version as 2.65git, the install exited after 'makeinfo --no-warn --no-split ../asymptote' in doc/png' without any messages but presumably with non-zero status.

**Comment by johncbowman** [comment]
This is a problem with the tiling code; I will look into it. For the record, here's a minimal example:

```
 import three;
settings.render=4;
settings.tex="pdflatex";
size(15cm,0);
draw(unitcylinder,red);
```

For now, try using a lower settings.render value.

**Comment by johncbowman** [comment]
The makeinfo error likely indicates that your version of makeinfo is obsolete.
This version of makeinfo works fine:

makeinfo --version
texi2any (GNU texinfo) 6.6

Copyright (C) 2017 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

**Comment by zarniwhoop73** [comment]
I am using a newer makeinfo on the machine where I tried 2.65git
$makeinfo --version
texi2any (GNU texinfo) 6.7

Copyright (C) 2017 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

As to a lower settings.render in my example, with 2 it works fine, with 4 it does not.

**Comment by zarniwhoop73** [comment]
I now cannot recreate the info problem when manually building and installing, although the installed info file definitely says it is still 2.64 (I went through the versions in order for the main issue). Maybe a weird one-off problem.

**Comment by zarniwhoop73** [comment]
"although the installed info file" : although the *original* installed info file

**Comment by johncbowman** [comment]
The tiling bug reported here is fixed in 1566850db2aac0b78ec9806e35af514e7a185e63.

**Comment by johncbowman** [comment]
Actually, the problem wasn't with the tiling code after all; it was due to a typo from 2015
in bf3be19f7f1daf5730dabbf5c89e8a4f0f451a7d. The proper fix is in 35428571776efdcaedcc9f920e927526849727a0 and the upcoming 2.66 release.
